subscribe.go

```diff
@@ -230,3 +230,52 @@ func (ls *legSubscriber) Close() error {
 	ls.cancel()
 	return nil
 }
+
+func (ls *legSubscriber) unlockOnce(ulOnce *sync.Once) {
+	ulOnce.Do(func() {
+		ls.syncmtx.Unlock()
+	})
+}
+
+func (ls *legSubscriber) Sync(ctx context.Context, p peer.ID, c cid.Cid) (chan cid.Cid, context.CancelFunc, error) {
+	ls.syncmtx.Lock()
+	out := make(chan cid.Cid)
+	var ulOnce sync.Once
+
+	v := Voucher{&c}
+	unsub := ls.transfer.t.SubscribeToEvents(ls.onSyncEvent(c, out, &ulOnce))
+	_, err := ls.transfer.t.OpenPullDataChannel(ctx, p, &v, c, legSelector(nil))
+	if err != nil {
+		log.Errorf("Error in data channel for sync: %v", err)
+		ls.syncmtx.Unlock()
```

Comment:
should this also be the `unlockOnce` call?

Reply:
It'd make no difference because we are not returning `CancelFunc` yet, so there is no option for someone to cancel and trying to unlock twice.